src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -891,21 +809,143 @@ EnumeratorResult createPatternBasedEnumeratorResult(ref ForEachEnumeratorInfo.Bu
                     return EnumeratorResult.Succeeded;
                 }
 
-                MethodSymbol getEnumeratorMethod = builder.GetEnumeratorMethod;
+                MethodSymbol getEnumeratorMethod = builder.GetEnumeratorInfo.Method;
                 diagnostics.Add(isAsync ? ErrorCode.ERR_BadGetAsyncEnumerator : ErrorCode.ERR_BadGetEnumerator, _syntax.Expression.Location, getEnumeratorMethod.ReturnType, getEnumeratorMethod);
                 return EnumeratorResult.FailedAndReported;
             }
+        }
+
+        private EnumeratorResult SatisfiesIEnumerableInterfaces(ref ForEachEnumeratorInfo.Builder builder, ref BoundExpression collectionExpr, bool isAsync, DiagnosticBag diagnostics, BoundExpression unwrappedCollectionExpr, TypeSymbol unwrappedCollectionExprType)
```

Comment:
> ref BoundExpression collectionExpr [](start = 106, length = 35)

It doesn't look like there is need to pass this parameter by ref. In fact, it doesn't look like there is a need to pass it at all because we assign to it before we use it.